php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -42,6 +61,84 @@ public ExpressionTree expression(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.STRING_LITERAL, token);
+  }
+
+  public LiteralTree booleanLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.BOOLEAN_LITERAL, token);
+  }
+
+  public LiteralTree nullLiteral(InternalSyntaxToken token) {
+   return new LiteralTreeImpl(Tree.Kind.NULL_LITERAL, token);
+  }
+
+  public LiteralTree magicConstantLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.MAGIC_CONSTANT, token);
+  }
+
+  public LiteralTree heredocLiteral(InternalSyntaxToken token) {
+    return new LiteralTreeImpl(Tree.Kind.HEREDOC_LITERAL, token);
+  }
+
+  public ExpandableStringCharactersTree expandableStringCharacters(InternalSyntaxToken token) {
+    return new ExpandableStringCharactersTreeImpl(token);
+  }
+
+  public VariableIdentifierTree expandableStringVariableIdentifier(InternalSyntaxToken token) {
+    return new VariableIdentifierTreeImpl(new IdentifierTreeImpl(token));
+  }
+
+  public IdentifierTree identifier(InternalSyntaxToken token) {
+    return new IdentifierTreeImpl(token);
+  }
+
+  public ArrayAccessTree expandableArrayAccess(InternalSyntaxToken openBracket, ExpressionTree offset, InternalSyntaxToken closeBracket) {
+    return new ArrayAccessTreeImpl(openBracket, offset, closeBracket);
+  }
+
+  public MemberAccessTree expandableObjectMemberAccess(InternalSyntaxToken arrow, IdentifierTree property) {
+    return new MemberAccessTreeImpl(arrow, property);
+  }
+
+  public ExpressionTree encapsulatedSimpleVar(VariableIdentifierTree variableIdentifier, Optional<ExpressionTree> partial) {
+    if (partial.isPresent()) {
+
+      if (partial.get() instanceof ArrayAccessTree) {
+        ((ArrayAccessTreeImpl) partial.get()).complete(variableIdentifier);
+      } else {
+        ((MemberAccessTreeImpl) partial.get()).complete(variableIdentifier);
+      }
+      return partial.get();
+    }
+
+    return variableIdentifier;
+  }
+
+  public ExpressionTree expressionRecovery(InternalSyntaxToken token) {
+    return new IdentifierTreeImpl(token);
+  }
+
+  public ExpressionTree encapsulatedSemiComplexVariable(InternalSyntaxToken openDollarCurly, ExpressionTree expressionTree, InternalSyntaxToken closeCurly) {
+    return new CompoundVariableTreeImpl(openDollarCurly, expressionTree, closeCurly);
+  }
+
+  public VariableIdentifierTree encapsulatedVariableIdentifier(InternalSyntaxToken spaces, InternalSyntaxToken variableIdentifier) {
+    return new VariableIdentifierTreeImpl(new IdentifierTreeImpl(variableIdentifier));
+  }
+
+  public ExpressionTree encapsulatedComplexVariable(InternalSyntaxToken openCurly, Tree lookahead, ExpressionTree expression, InternalSyntaxToken closeCurly) {
+    return new ComputedVariableTreeImpl(openCurly, expression, closeCurly);
+  }
+
+  public ExpandableStringLiteralTree expandableStringLiteral(Tree spacing, InternalSyntaxToken openDoubleQuote, List<ExpressionTree> expressions, InternalSyntaxToken closeDoubleQuote) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 185 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)